Dockerfile

```diff
@@ -22,9 +22,9 @@ COPY controllers/ controllers/
 # the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
 # by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
 USER root
-RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager main.go
```

Comment:
should we leave the GOOS param since it already has the same defaulted value of `linux`, and update the GOARCH similarly to use $TARGETARCH but default to `amd64`?  If not, should we remove the ARG layers since they're now unused?

Reply:
Good point. I didn't notice that this was different compared to the Konflux Dockerfiles. I guess I should have read the Code Rabbit review comment more closely lol.